homeassistant/components/lamarzocco/config_flow.py

```diff
@@ -153,6 +154,13 @@ async def async_step_machine_selection(
                     self._config[CONF_HOST] = user_input[CONF_HOST]
 
             if not errors:
+                if self.reconfigure_entry:
+                    for service_info in async_discovered_service_info(self.hass):
+                        self._discovered[service_info.name] = service_info.address
+
+                    if self._discovered:
```

Comment:
Should we have a check here to only display this step if there is more than one device to choose from?

Reply:
this should be false if `_discovered` is empty, so already done. Or am I missing something?

Reply:
I explicitly didn't want to filter on name or `ids` here as this is meant to be a fallback option in case your machine is out of line somehow

Reply:
`self._discovered` is a dict with zero, one or multiple of `service_info`.
So can there be only one key in the dict and then it's known which it is and then we could skip the step.

e.g.
```python
if self._discovered and len(self._discovered) > 1:
```

Reply:
if there is only one device it could be a random BT device that is not a `LaMarzocco` machine, so I'd rather not choose it automatically. That's what I meant with the previous comment.